site/content/docs/agentic/ray-mcp/index.md

```diff
@@ -0,0 +1,475 @@
+```
+
+Which should display output similar to:
+```bash
+NAME                   DATA   AGE
+kube-root-ca.crt       1      4h47m
+llama-chat-templates   1      3h18m
+```
+### Test the Ray Serve Deployment
+
+Test the model by setting up port forwarding and sending requests.
+
+Set up port forwarding to the Ray Serve endpoint:
+
+```bash
+kubectl port-forward service/llama-31-8b-serve-svc 8000:8000
+```
+
+Send a weather query that requires tool usage:
+
+```bash
+curl http://127.0.0.1:8000/v1/chat/completions \
+-X POST \
+-H "Content-Type: application/json" \
+-d '{
+    "model": "meta-llama/Llama-3.1-8B-Instruct",
+    "messages": [
+        {
+          "role": "user",
+          "content": "What is the weather in New York today?"
+        }
+    ],
+    "tools": [
+        {
+          "type": "function",
+          "function": {
+            "name": "get_weather",
+            "description": "Get the current weather for a city",
+            "parameters": {
+              "type": "object",
+              "properties": {
+                "city": {
+                  "type": "string",
+                  "description": "The city name"
+                }
+              },
+              "required": ["city"]
+            }
+          }
+        }
+    ]
+}' | jq
+```
+
+The response should include a tool call to `get_weather` with the parameter `city: "New York"`, and the `"finish_reason"` should be `"tool_calls"`. And the response should contain something like this:
+
+```log
+{
+  "id": "chatcmpl-a7569e70-c904-40cb-9aad-a1c5944fc503",
+  "object": "chat.completion",
+  "created": 1749044680,
+  "model": "meta-llama/Llama-3.1-8B-Instruct",
+  "choices": [
+    {
+      "index": 0,
+      "message": {
+        "role": "assistant",
+        "reasoning_content": null,
+        "content": null,
+        "tool_calls": [
+          {
+            "id": "chatcmpl-tool-7d85f9313e2e4d76904a3ae2d76c69ac",
+            "type": "function",
+            "function": {
+              "name": "get_weather",
+              "arguments": "{\"city\": \"New York\"}"
+            }
+          }
+        ]
+      },
+      "logprobs": null,
+      "finish_reason": "tool_calls",
+      "stop_reason": 128008
+    }
+  ],
+  "usage": {
+    "prompt_tokens": 254,
+    "total_tokens": 273,
+    "completion_tokens": 19,
+    "prompt_tokens_details": null
+  },
+  "prompt_logprobs": null
+}
+```
+
+## Step 3: Deploy the MCP server
+
+Navigate to the MCP Server directory:
+
+```bash
+cd ../mcp_server
+```
+
+Build and push the MCP Server container image:
+
+```bash
+gcloud builds submit \
+    --tag us-docker.pkg.dev/$PROJECT_ID/llama-ray-mcp/mcp-server:latest \
+    --project=$PROJECT_ID \
+    .
+```
+
+Update the `./deployment_weather_mcp.yaml` file `<PROJECT_ID>` placeholders where applicable. Apply the manifest:
+```bash
+kubectl apply -f deployment_weather_mcp.yaml
+```
+
+Validate your MCP Server by running this command:
+```bash
+npx @modelcontextprotocol/inspector
```

Comment:
Should we also add nodejs in the [list of required tools](https://github.com/ai-on-gke/website/blob/adk-ray-mcp/site/content/docs/agentic/ray-mcp/index.md#before-you-begin)?